cpp/src/arrow/compute/kernels/scalar_compare.cc

```diff
@@ -439,6 +472,329 @@ struct ScalarMinMax {
+        continue;
+      }
+      util::optional<string_view> result = valid_cols.front();
+      for (size_t col = 1; col < batch.values.size(); ++col) {
+        util::optional<string_view> value = valid_cols[col];
+        if (!value) {
+          DCHECK(options.skip_nulls);
+          continue;
+        }
+        result = !result ? *value : Op::Call(*result, *value);
+      }
+      if (result) {
+        builder.UnsafeAppend(*result);
+      } else {
+        builder.UnsafeAppendNull();
+      }
+    }
+
+    std::shared_ptr<Array> string_array;
+    RETURN_NOT_OK(builder.Finish(&string_array));
+    *out = *string_array->data();
+    out->mutable_array()->type = batch[0].type();
+    DCHECK_EQ(batch.length, out->array()->length);
+    DCHECK_GE(final_size,
+              checked_cast<const ArrayType&>(*string_array).total_values_length());
+    return Status::OK();
+  }
+
+  // Compute the length of the output for the given position, or -1 if it would be null.
+  static int64_t CalculateRowSize(const ElementWiseAggregateOptions& options,
+                                  const ExecBatch& batch, const int64_t index) {
+    const auto num_args = batch.values.size();
+    int64_t final_size = 0;
+    for (size_t i = 0; i < num_args; i++) {
+      int64_t element_size = 0;
+      bool valid = true;
+      if (batch[i].is_scalar()) {
+        const auto& scalar = *batch[i].scalar();
+        valid = scalar.is_valid;
+        element_size = static_cast<int64_t>(UnboxScalar<Type>::Unbox(scalar).size());
+      } else {
+        const ArrayData& array = *batch[i].array();
+        valid = !array.MayHaveNulls() ||
+                bit_util::GetBit(array.buffers[0]->data(), array.offset + index);
+        const offset_type* offsets = array.GetValues<offset_type>(1);
+        element_size = offsets[index + 1] - offsets[index];
+      }
+      if (!valid) {
+        if (options.skip_nulls) {
+          continue;
+        }
+        return -1;
+      }
+      // Conservative estimation of the element size.
+      final_size = std::max(final_size, element_size);
+    }
+    return final_size;
+  }
+};
+
+template <typename Op>
+struct FixedSizeBinaryScalarMinMax {
+  static Status Exec(KernelContext* ctx, const ExecBatch& batch, Datum* out) {
+    const ElementWiseAggregateOptions& options = MinMaxState::Get(ctx);
+    if (std::all_of(batch.values.begin(), batch.values.end(),
+                    [](const Datum& d) { return d.is_scalar(); })) {
+      return ExecOnlyScalar(ctx, options, batch, out);
+    }
+    return ExecContainingArrays(ctx, options, batch, out);
+  }
+
+  static Status ExecOnlyScalar(KernelContext* ctx,
+                               const ElementWiseAggregateOptions& options,
+                               const ExecBatch& batch, Datum* out) {
+    if (batch.values.empty()) {
+      return Status::OK();
+    }
+    BaseBinaryScalar* output = checked_cast<BaseBinaryScalar*>(out->scalar().get());
```

Comment:
There are a few places where you can use `auto` to simplify the declared types.

Reply:
Done.